src/utils/pusher.tsx

```diff
@@ -80,19 +92,10 @@ import React from "react";
 export const PusherProvider: React.FC<
   React.PropsWithChildren<{ slug: string }>
 > = ({ slug, children }) => {
-  const store = React.useMemo<StoreApi<PusherZustandStore>>(
-    () => createPusherStore(slug),
-    [slug]
-  );
-
-  React.useEffect(() => {
-    return () => {
-      store.getState().pusherClient.disconnect();
-    };
-  }, [store]);
+  const store = usePusherStore(slug);
```

Comment:
Naming here is a tad confusing

Reply:
Yeah I agree, can definitely name this better.